src/rust/engine/process_execution/remote/src/remote.rs

```diff
@@ -173,19 +175,25 @@ impl CommandRunner {
     execution_concurrency_limit: usize,
     capabilities_cell_opt: Option<Arc<OnceCell<ServerCapabilities>>>,
   ) -> Result<Self, String> {
-    let execution_use_tls = execution_address.starts_with("https://");
+    let needs_tls =
+      execution_address.starts_with("https://") || execution_address.starts_with("grpcs://");
```

Comment:
A check for `grpcs://` is unnecessary here since the Python layer translates `grpcs` to `https`. See https://github.com/pantsbuild/pants/blob/eecd07908868af3e3e0d97a91852de2785b7b16c/src/python/pants/option/global_options.py#L467-L471.

Reply:
Hmm, I see your point, but it *looks* like a bug when we don't check it -- I know what I wrote in the config, and this doesn't check it. I'm wondering whether that rewrite should happen at the last possible moment, not far far away from where the value is processed?

Reply:
(This is a particular issue of stringly typed interfaces maybe. We don't care about the "protocol" spelling, we care about whether it's TLS. Maybe we should convert to a URI, then it'd be an enumeration so you can't write "wrong" code.)

Reply:
Strongly typing this is fine by me.

Reply:
This protocol/address code is changing a lot with https://github.com/pantsbuild/pants/pull/19827, I've added tweaking this to https://github.com/pantsbuild/pants/issues/19902.

Reply:
@huonw Great, will revert and we'll tackle it separately to not scope-creep this PR.